module/VuFindConsole/src/VuFindConsole/Command/Util/BrowscapCommand.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+/**
+ * Console command: browscap
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Console
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+
+namespace VuFindConsole\Command\Util;
+
+use Laminas\Cache\Psr\SimpleCache\SimpleCacheDecorator;
+use Psr\Log\LogLevel;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Logger\ConsoleLogger;
+use Symfony\Component\Console\Output\OutputInterface;
+use VuFind\Cache\Manager as CacheManager;
+use VuFind\Http\GuzzleService;
+
+/**
+ * Console command: browscap
+ *
+ * @category VuFind
+ * @package  Console
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class BrowscapCommand extends Command
```

Comment:
This prompted me to do a bit of work on the cache manager so that initialization of caches is deferred until they're actually needed. I've wanted to do this for a long time since it can save a few cycles for most requests. What do you think? If this is too radical departure from past, I can change it so that other caches are initialized right away, but I'd really like to keep it this way.

What's unfortunate is that Laminas cache requires write access to the directory and its files even if the cache is marked as `['writable' => false]`, read access is not enough.

Reply:
Seems reasonable to me, and #3400 is now merged. I'll test all this again after conflicts are merged, and when I have a little more time (in other words, next week :-) ).